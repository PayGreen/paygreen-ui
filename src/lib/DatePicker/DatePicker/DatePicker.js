import React, { useState } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { DateContextProvider } from '../context/DateContext';
import DatePickerBase from './style';
import DropDown from '../../Dropdown/Dropdown';
import DatePickerInput from '../DatePickerInput/DatePickerInput';
import Popin from '../../Popin/Popin';
import Calendar from '../Calendar/Calendar';
import {
    buttonSizeDefault,
    buttonSizeOptions,
    formStatusDefault,
    formStatusOptions,
    inputWidthDefault,
    inputWidthOptions,
} from '../../../shared/constants';

const DatePicker = ({
    readOnly,
    disabled,
    fieldSize,
    blockWidth,
    hasHelpButton,
    locale,
    minimumDate,
    maximumDate,
    colorStatus,
    ...rest
}) => {
    const [selectedDate, setSelectedDate] = useState(moment());

    return (
        <DateContextProvider value={[selectedDate, setSelectedDate]}>
            <DatePickerBase {...rest}>
                <DropDown {...rest}>
                    <DatePickerInput
                        value={selectedDate.format('DD/MM/YYYY')}
                        readOnly={readOnly}
                        disabled={disabled}
                        fieldSize={fieldSize}
                        blockWidth={blockWidth}
                        hasHelpButton={hasHelpButton}
                        {...rest}
                    />

                    <Popin {...rest}>
                        <Calendar
                            currentMonth={selectedDate.month()}
                            locale={locale}
                            minimumDate={minimumDate}
                            maximumDate={maximumDate}
                            colorStatus={colorStatus}
                            {...rest}
                        />
                    </Popin>
                </DropDown>
            </DatePickerBase>
        </DateContextProvider>
    );
};

DatePicker.propTypes = {
    // Input props
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    fieldSize: PropTypes.oneOf(Object.values(buttonSizeOptions)),
    blockWidth: PropTypes.oneOf(Object.values(inputWidthOptions)),
    hasHelpButton: PropTypes.bool,
    inputRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.shape({ current: PropTypes.any }),
    ]),

    // Calendar props
    locale: PropTypes.string,
    minimumDate: PropTypes.object, // Moment instance
    maximumDate: PropTypes.object, // Moment instance
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
};

DatePicker.defaultProps = {
    // Input props
    disabled: false,
    readOnly: false,
    fieldSize: buttonSizeDefault,
    blockWidth: inputWidthDefault,
    hasHelpButton: false,

    // Calendar props
    locale: 'fr',
    minimumDate: null,
    maximumDate: null,
    colorStatus: formStatusDefault,
};

export default DatePicker;
