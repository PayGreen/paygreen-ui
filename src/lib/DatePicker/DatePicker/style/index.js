import styled from 'styled-components';

const DatePickerBase = styled.div``;

export default DatePickerBase;
