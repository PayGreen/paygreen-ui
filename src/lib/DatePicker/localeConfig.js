const config = {
    'pg-fr': {
        months: 'janvier_février_mars_avril_mai_juin_juillet_août_septembre_octobre_novembre_décembre'.split(
            '_',
        ),
        monthsParseExact: true,
        weekdays: 'dimanche_lundi_mardi_mercredi_jeudi_vendredi_samedi'.split(
            '_',
        ),
        weekdaysParseExact: true,
        longDateFormat: {
            L: 'DD/MM/YYYY',
        },
        week: {
            dow: 1, // Monday is the first day of the week.
            doy: 4, // Used to determine first week of the year.
        },
    },
    'pg-en': {
        months: 'january_february_march_april_may_june_july_august_september_october_november_december'.split(
            '_',
        ),
        monthsParseExact: true,
        weekdays: 'sunday_monday_tuesday_wednesday_thursday_friday_saturday'.split(
            '_',
        ),
        weekdaysParseExact: true,
        longDateFormat: {
            L: 'DD/MM/YYYY',
        },
        week: {
            dow: 1, // Monday is the first day of the week.
            doy: 4, // Used to determine first week of the year.
        },
    },
};

export default config;
