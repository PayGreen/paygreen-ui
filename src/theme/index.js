import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
