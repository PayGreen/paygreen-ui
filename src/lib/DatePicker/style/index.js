import styled from 'styled-components';
import { DaInputBase } from '../../DaInput/style';
import { PopinBase } from '../../Popin/style';

const DatePickerBase = styled.div`
    ${DaInputBase} {
        max-width: ${props => props.theme.form.inputWidth.xxs};
    }

    ${PopinBase} {
        width: auto;
        font-size: ${props => props.theme.font.size.sm};
    }
`;

export { DatePickerBase };
