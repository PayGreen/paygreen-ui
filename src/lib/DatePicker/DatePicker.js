import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import {
    inputWidthOptions,
    buttonSizeOptions,
    formStatusDefault,
    formStatusOptions,
    localeOptions,
    localeDefault,
} from '../../shared/constants';
import DropDown from '../Dropdown/Dropdown';
import DaInput from '../DaInput/DaInput';
import Popin from '../Popin/Popin';
import { DateContextProvider } from './context/DateContext';
import Calendar from './Calendar/Calendar';
import localeConfig from './localeConfig';
import { DatePickerBase } from './style';

const DatePicker = ({
    colorStatus,
    locale,
    minimumDate,
    maximumDate,
    resetDate,
    value,
    onChange,
    ...rest
}) => {
    // Update moment locale globally
    // Note : Default locale of moment is 'en', overwrite it with custom config 'pg-fr'
    const pgLocale = localeConfig['pg-' + localeOptions[locale]]
        ? 'pg-' + localeOptions[locale]
        : 'pg-' + localeDefault;
    const dateFormat = localeConfig[pgLocale].longDateFormat.L;
    moment.updateLocale(pgLocale, localeConfig[pgLocale]);

    // Save default value of input...
    const [selectedDate, setSelectedDate] = useState(
        moment(value, dateFormat, true).isValid()
            ? moment(value, dateFormat)
            : null,
    );

    // And extract input state handling
    const [inputValue, setInputValue] = useState(value);

    useEffect(() => {
        if (selectedDate) {
            setInputValue(selectedDate.format(dateFormat));
        }
    }, [selectedDate]);

    // To set date value via Input change
    const setInputDate = value => {
        setSelectedDate(moment(value, dateFormat));
        onChange(value);
    };

    /**
     * @description - to create condition to validate or invalidate date value
     * @param {string} dateValue - date received, format DD/MM/YYYY
     * @param {string} dateReference - date reference to decide if dateValue is correct minimumDate | maximumDate, format DD/MM/YYYY
     * @param {string} position - to determinate which method from Moment to use based on date's position 'isBefore' | 'isAfter'
     * @returns {boolean}
     */
    const isValidDate = (dateValue, dateReference, position) => {
        return (
            moment(dateValue, dateFormat, true).isValid() &&
            dateReference &&
            moment(dateReference, dateFormat, true).isValid() &&
            (position === 'isBefore'
                ? moment(dateValue, dateFormat).isBefore(
                      moment(dateReference, dateFormat),
                  )
                : moment(dateValue, dateFormat).isAfter(
                      moment(dateReference, dateFormat),
                  ))
        );
    };

    /**
     * @description - to reset wrong date value via Input based on minimumDate, maximumDate and resetDate provided
     * @param {string} dateValue - date received, format DD/MM/YYYY
     */
    const resetWrongDate = dateValue => {
        if (isValidDate(dateValue, maximumDate, 'isAfter')) {
            setInputDate(maximumDate);
        } else if (isValidDate(dateValue, minimumDate, 'isBefore')) {
            setInputDate(minimumDate);
        } else {
            const defaultDate =
                resetDate && moment(resetDate, dateFormat, true).isValid()
                    ? resetDate
                    : moment().format(dateFormat);
            setInputDate(defaultDate);
        }
    };

    // Handle value change via Input and prevent final selectedDate to be applied if incorrect (in format or in value compared to minimumDate and maximumDate)
    const handleInputOnChange = e => {
        if (selectedDate) {
            setSelectedDate(null);
        }

        setInputValue(e.target.value);

        if (
            isValidDate(e.target.value, maximumDate, 'isBefore') &&
            isValidDate(e.target.value, minimumDate, 'isAfter')
        ) {
            setInputDate(e.target.value);
        } else if (
            !minimumDate &&
            isValidDate(e.target.value, maximumDate, 'isBefore')
        ) {
            setInputDate(e.target.value);
        } else if (
            !maximumDate &&
            isValidDate(e.target.value, minimumDate, 'isAfter')
        ) {
            setInputDate(e.target.value);
        } else if (
            !minimumDate &&
            !maximumDate &&
            moment(e.target.value, dateFormat, true).isValid()
        ) {
            setInputDate(e.target.value);
        } else if (e.target.value.indexOf('_') >= 0) {
            // we don't do anything if date is not complete and input mask contains '_' characters
            return;
        } else {
            resetWrongDate(e.target.value);
        }
    };

    // Handle value change via CalendarCell Buttons
    const handleButtonOnChange = e => {
        onChange(moment(e).format(dateFormat));
    };

    const calcMonthIndex = () => {
        if (!selectedDate || selectedDate === moment().startOf('d')) {
            return moment().month();
        }

        const yearDiff = selectedDate.format('YYYY') - moment().format('YYYY');
        const monthDiff = selectedDate.format('M') - moment().format('M');

        return yearDiff !== 0
            ? moment().month() + yearDiff * 12 + monthDiff
            : moment().month() + monthDiff;
    };

    return (
        <DateContextProvider
            value={[selectedDate, setSelectedDate, setInputValue]}
        >
            <DatePickerBase theme={rest.theme}>
                <DropDown theme={rest.theme}>
                    <DaInput
                        {...rest}
                        mask="99/99/9999"
                        type="text"
                        value={inputValue}
                        onChange={handleInputOnChange}
                    />

                    {rest.readOnly || rest.disabled ? null : (
                        <Popin theme={rest.theme}>
                            <Calendar
                                theme={rest.theme}
                                currentMonth={calcMonthIndex()}
                                minimumDate={
                                    moment(
                                        minimumDate,
                                        dateFormat,
                                        true,
                                    ).isValid()
                                        ? moment(minimumDate, dateFormat)
                                        : null
                                }
                                maximumDate={
                                    moment(
                                        maximumDate,
                                        dateFormat,
                                        true,
                                    ).isValid()
                                        ? moment(maximumDate, dateFormat)
                                        : null
                                }
                                colorStatus={colorStatus}
                                handleOnChange={handleButtonOnChange}
                            />
                        </Popin>
                    )}
                </DropDown>
            </DatePickerBase>
        </DateContextProvider>
    );
};

DatePicker.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,

    // Input props
    placeholder: PropTypes.string,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    isRounded: PropTypes.bool,
    fieldSize: PropTypes.oneOf(Object.values(buttonSizeOptions)),
    blockWidth: PropTypes.oneOf(Object.values(inputWidthOptions)),
    hasHelpButton: PropTypes.bool,
    inputRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.shape({ current: PropTypes.any }),
    ]),

    // Calendar props
    locale: PropTypes.oneOf(Object.values(localeOptions)), // 2-letters short language code
    minimumDate: PropTypes.string,
    maximumDate: PropTypes.string,
    resetDate: PropTypes.string,
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
};

DatePicker.defaultProps = {
    value: '',

    // Input props
    disabled: false,
    readOnly: false,

    // Calendar props
    locale: localeDefault,
    minimumDate: null,
    maximumDate: null,
    resetDate: '',
    colorStatus: formStatusDefault,
};

export default DatePicker;
