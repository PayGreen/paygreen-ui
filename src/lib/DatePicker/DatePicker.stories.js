import React, { useState } from 'react';
import moment from 'moment';
import { storiesOf } from '@storybook/react';
import {
    withKnobs,
    text,
    select,
    radios,
    boolean,
} from '@storybook/addon-knobs';
import {
    folder,
    buttonSizeDefault,
    buttonSizeOptions,
    formStatusDefault,
    formStatusOptions,
    localeDefault,
    localeOptions,
} from '../../shared/constants';
import DatePicker from './DatePicker';

storiesOf(folder.form + 'DatePicker', module)
    .addDecorator(withKnobs)
    .add('DatePicker', () =>
        React.createElement(() => {
            const [date, setDate] = useState();

            return (
                <DatePicker
                    value={date}
                    onChange={date => setDate(date)}
                    placeholder={'Ex: ' + moment().format('DD/MM/YYYY')}
                    disabled={boolean('Disabled', false)}
                    readOnly={boolean('Readonly', false)}
                    locale={select('Locale', localeOptions, localeDefault)}
                    isRounded={boolean('Is rounded', false)}
                    fieldSize={radios(
                        'Field size',
                        buttonSizeOptions,
                        buttonSizeDefault,
                    )}
                    colorStatus={select(
                        'Color status',
                        formStatusOptions,
                        formStatusDefault,
                    )}
                    minimumDate={text(
                        'Minimum date',
                        moment().add(-1, 'M').format('DD/MM/YYYY'),
                    )}
                    maximumDate={text(
                        'Maximum date',
                        moment().add(1, 'M').format('DD/MM/YYYY'),
                    )}
                    resetDate={text(
                        'Reset date',
                        moment().format('DD/MM/YYYY'),
                    )}
                />
            );
        }),
    );
