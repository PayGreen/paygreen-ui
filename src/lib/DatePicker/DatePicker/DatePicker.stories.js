import React from 'react';
import moment from 'moment';
import { storiesOf } from '@storybook/react';
import {
    withKnobs,
    text,
    select,
    radios,
    boolean,
} from '@storybook/addon-knobs';
import {
    folder,
    buttonSizeDefault,
    buttonSizeOptions,
    inputWidthDefault,
    inputWidthOptions,
    formStatusDefault,
    formStatusOptions,
} from '../../../shared/constants';
import DatePicker from './DatePicker';

storiesOf(folder.form + folder.sub.datePicker + 'DatePicker', module)
    .addDecorator(withKnobs)
    .add('DatePicker', () => (
        <DatePicker
            placeholder={text('Placeholder', moment().format('DD/MM/YYYY'))}
            disabled={boolean('Disabled', false)}
            readOnly={boolean('Readonly', false)}
            fieldSize={radios(
                'Field size',
                buttonSizeOptions,
                buttonSizeDefault,
            )}
            blockWidth={radios(
                'Input width',
                inputWidthOptions,
                inputWidthDefault,
            )}
            hasHelpButton={boolean('Help button', false)}
            colorStatus={select(
                'Color status',
                formStatusOptions,
                formStatusDefault,
            )}
        />
    ));
