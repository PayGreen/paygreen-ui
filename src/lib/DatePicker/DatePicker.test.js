import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import DatePicker from './DatePicker';

it('renders without crashing', () => {
    const datePicker = TestRenderer.create(
        <DatePicker
            theme={ThemeDefault}
            value="23/04/2021"
            onChange={() => {}}
        />,
    );
    expect(datePicker.toJSON()).toMatchSnapshot();
});
