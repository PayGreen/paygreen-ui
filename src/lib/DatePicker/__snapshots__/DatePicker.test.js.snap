// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__DatePickerBase-w9sfup-0 fhxaqn"
>
  <div
    className="style__DropdownBase-ht2epv-0 dYNCxh"
  >
    <div
      className="style__DaInputBase-sc-13xli4f-0 flOZAF"
    >
      <input
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        readOnly={false}
        type="text"
        value="23/04/2021"
      />
    </div>
    <div
      className="style__PopinBase-sc-1ky0zqo-0 kIQbUy"
    >
      <div
        className="style__CalendarNavbarBase-sc-9kauge-0 eMrxQk"
      >
        <button
          className="style__IconBase-ak3679-0 KOHwy icon"
          onClick={[Function]}
          type="button"
        >
          <svg
            viewBox="0 0 143 143"
          >
            <path
              d="M97.8 123.9c-.8 0-1.6-.3-2.3-.9l-55-47.6c-.8-.7-1.2-1.6-1.2-2.6s.4-2 1.2-2.6l55-48.2c1.5-1.3 3.7-1.1 4.9.3 1.3 1.5 1.1 3.7-.3 4.9l-52 45.6 52 45c1.5 1.3 1.6 3.5.4 4.9-.8.8-1.8 1.2-2.7 1.2z"
            />
          </svg>
        </button>
        avril 2021
        <button
          className="style__IconBase-ak3679-0 KOHwy icon"
          onClick={[Function]}
          type="button"
        >
          <svg
            viewBox="0 0 143 143"
          >
            <path
              d="M43.5 123.9c-1 0-1.9-.4-2.6-1.2-1.3-1.5-1.1-3.7.3-4.9l52-45.6-52-45c-1.5-1.3-1.6-3.5-.4-4.9 1.3-1.5 3.5-1.6 4.9-.4l55 47.6c.8.7 1.2 1.6 1.2 2.6s-.4 2-1.2 2.6l-55 48.2c-.6.7-1.4 1-2.2 1z"
            />
          </svg>
        </button>
      </div>
      <div
        className="style__CalendarWeekdaysBase-sc-1ai2hag-0 krhkQ"
      >
        <div>
          l
        </div>
        <div>
          m
        </div>
        <div>
          m
        </div>
        <div>
          j
        </div>
        <div>
          v
        </div>
        <div>
          s
        </div>
        <div>
          d
        </div>
      </div>
      <div
        className="style__CalendarGridBase-sc-1qui6a5-0 ehqGoy"
      >
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          10
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          11
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          12
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          13
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          14
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          15
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          16
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          17
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          18
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          19
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          20
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          21
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          22
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 kqyGos"
          onClick={[Function]}
          type="button"
        >
          23
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          24
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          25
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          26
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          27
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          28
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          29
        </button>
        <button
          className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
          onClick={[Function]}
          type="button"
        >
          30
        </button>
      </div>
    </div>
  </div>
</div>
`;
